// @aws-sdk/client-elastic-load-balancing-v2/dist-cjs/index.js
var __exportStar = (m, e) => { Object.assign(e, m); };
const { getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, Command, createWaiter, checkExceptions, WaiterState, createAggregatedClient } = require("@smithy/core/client");
exports.$Command = Command;
exports.__Client = Client;
const { resolveRegionConfig } = require("@smithy/core/config");
const { resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveHttpAuthSchemeConfig, defaultElasticLoadBalancingV2HttpAuthSchemeParametersProvider } = require("./auth/httpAuthSchemeProvider");
const { getRuntimeConfig } = require("./runtimeConfig");
const { AddListenerCertificates$, AddTags$, AddTrustStoreRevocations$, CreateListener$, CreateLoadBalancer$, CreateRule$, CreateTargetGroup$, CreateTrustStore$, DeleteListener$, DeleteLoadBalancer$, DeleteRule$, DeleteSharedTrustStoreAssociation$, DeleteTargetGroup$, DeleteTrustStore$, DeregisterTargets$, DescribeAccountLimits$, DescribeCapacityReservation$, DescribeListenerAttributes$, DescribeListenerCertificates$, DescribeListeners$, DescribeLoadBalancerAttributes$, DescribeLoadBalancers$, DescribeRules$, DescribeSSLPolicies$, DescribeTags$, DescribeTargetGroupAttributes$, DescribeTargetGroups$, DescribeTargetHealth$, DescribeTrustStoreAssociations$, DescribeTrustStoreRevocations$, DescribeTrustStores$, GetResourcePolicy$, GetTrustStoreCaCertificatesBundle$, GetTrustStoreRevocationContent$, ModifyCapacityReservation$, ModifyIpPools$, ModifyListenerAttributes$, ModifyListener$, ModifyLoadBalancerAttributes$, ModifyRule$, ModifyTargetGroupAttributes$, ModifyTargetGroup$, ModifyTrustStore$, RegisterTargets$, RemoveListenerCertificates$, RemoveTags$, RemoveTrustStoreRevocations$, SetIpAddressType$, SetRulePriorities$, SetSecurityGroups$, SetSubnets$ } = require("./schemas/schemas_0");
__exportStar(require("./schemas/schemas_0"), exports);
__exportStar(require("./models/errors"), exports);
const { ElasticLoadBalancingV2ServiceException } = require("./models/ElasticLoadBalancingV2ServiceException");
exports.ElasticLoadBalancingV2ServiceException = ElasticLoadBalancingV2ServiceException;

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "elasticloadbalancing",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class ElasticLoadBalancingV2Client extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultElasticLoadBalancingV2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AddListenerCertificatesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "AddListenerCertificatesCommand")
    .sc(AddListenerCertificates$)
    .build() {
}

class AddTagsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddTags", {})
    .n("ElasticLoadBalancingV2Client", "AddTagsCommand")
    .sc(AddTags$)
    .build() {
}

class AddTrustStoreRevocationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "AddTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "AddTrustStoreRevocationsCommand")
    .sc(AddTrustStoreRevocations$)
    .build() {
}

class CreateListenerCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateListener", {})
    .n("ElasticLoadBalancingV2Client", "CreateListenerCommand")
    .sc(CreateListener$)
    .build() {
}

class CreateLoadBalancerCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateLoadBalancer", {})
    .n("ElasticLoadBalancingV2Client", "CreateLoadBalancerCommand")
    .sc(CreateLoadBalancer$)
    .build() {
}

class CreateRuleCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateRule", {})
    .n("ElasticLoadBalancingV2Client", "CreateRuleCommand")
    .sc(CreateRule$)
    .build() {
}

class CreateTargetGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "CreateTargetGroupCommand")
    .sc(CreateTargetGroup$)
    .build() {
}

class CreateTrustStoreCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "CreateTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "CreateTrustStoreCommand")
    .sc(CreateTrustStore$)
    .build() {
}

class DeleteListenerCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteListener", {})
    .n("ElasticLoadBalancingV2Client", "DeleteListenerCommand")
    .sc(DeleteListener$)
    .build() {
}

class DeleteLoadBalancerCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteLoadBalancer", {})
    .n("ElasticLoadBalancingV2Client", "DeleteLoadBalancerCommand")
    .sc(DeleteLoadBalancer$)
    .build() {
}

class DeleteRuleCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteRule", {})
    .n("ElasticLoadBalancingV2Client", "DeleteRuleCommand")
    .sc(DeleteRule$)
    .build() {
}

class DeleteSharedTrustStoreAssociationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteSharedTrustStoreAssociation", {})
    .n("ElasticLoadBalancingV2Client", "DeleteSharedTrustStoreAssociationCommand")
    .sc(DeleteSharedTrustStoreAssociation$)
    .build() {
}

class DeleteTargetGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "DeleteTargetGroupCommand")
    .sc(DeleteTargetGroup$)
    .build() {
}

class DeleteTrustStoreCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeleteTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "DeleteTrustStoreCommand")
    .sc(DeleteTrustStore$)
    .build() {
}

class DeregisterTargetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DeregisterTargets", {})
    .n("ElasticLoadBalancingV2Client", "DeregisterTargetsCommand")
    .sc(DeregisterTargets$)
    .build() {
}

class DescribeAccountLimitsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeAccountLimits", {})
    .n("ElasticLoadBalancingV2Client", "DescribeAccountLimitsCommand")
    .sc(DescribeAccountLimits$)
    .build() {
}

class DescribeCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeCapacityReservation", {})
    .n("ElasticLoadBalancingV2Client", "DescribeCapacityReservationCommand")
    .sc(DescribeCapacityReservation$)
    .build() {
}

class DescribeListenerAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListenerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenerAttributesCommand")
    .sc(DescribeListenerAttributes$)
    .build() {
}

class DescribeListenerCertificatesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenerCertificatesCommand")
    .sc(DescribeListenerCertificates$)
    .build() {
}

class DescribeListenersCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeListeners", {})
    .n("ElasticLoadBalancingV2Client", "DescribeListenersCommand")
    .sc(DescribeListeners$)
    .build() {
}

class DescribeLoadBalancerAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeLoadBalancerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeLoadBalancerAttributesCommand")
    .sc(DescribeLoadBalancerAttributes$)
    .build() {
}

class DescribeLoadBalancersCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeLoadBalancers", {})
    .n("ElasticLoadBalancingV2Client", "DescribeLoadBalancersCommand")
    .sc(DescribeLoadBalancers$)
    .build() {
}

class DescribeRulesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeRules", {})
    .n("ElasticLoadBalancingV2Client", "DescribeRulesCommand")
    .sc(DescribeRules$)
    .build() {
}

class DescribeSSLPoliciesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeSSLPolicies", {})
    .n("ElasticLoadBalancingV2Client", "DescribeSSLPoliciesCommand")
    .sc(DescribeSSLPolicies$)
    .build() {
}

class DescribeTagsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTags", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTagsCommand")
    .sc(DescribeTags$)
    .build() {
}

class DescribeTargetGroupAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetGroupAttributes", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetGroupAttributesCommand")
    .sc(DescribeTargetGroupAttributes$)
    .build() {
}

class DescribeTargetGroupsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetGroups", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetGroupsCommand")
    .sc(DescribeTargetGroups$)
    .build() {
}

class DescribeTargetHealthCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTargetHealth", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTargetHealthCommand")
    .sc(DescribeTargetHealth$)
    .build() {
}

class DescribeTrustStoreAssociationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStoreAssociations", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoreAssociationsCommand")
    .sc(DescribeTrustStoreAssociations$)
    .build() {
}

class DescribeTrustStoreRevocationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoreRevocationsCommand")
    .sc(DescribeTrustStoreRevocations$)
    .build() {
}

class DescribeTrustStoresCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "DescribeTrustStores", {})
    .n("ElasticLoadBalancingV2Client", "DescribeTrustStoresCommand")
    .sc(DescribeTrustStores$)
    .build() {
}

class GetResourcePolicyCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetResourcePolicy", {})
    .n("ElasticLoadBalancingV2Client", "GetResourcePolicyCommand")
    .sc(GetResourcePolicy$)
    .build() {
}

class GetTrustStoreCaCertificatesBundleCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetTrustStoreCaCertificatesBundle", {})
    .n("ElasticLoadBalancingV2Client", "GetTrustStoreCaCertificatesBundleCommand")
    .sc(GetTrustStoreCaCertificatesBundle$)
    .build() {
}

class GetTrustStoreRevocationContentCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "GetTrustStoreRevocationContent", {})
    .n("ElasticLoadBalancingV2Client", "GetTrustStoreRevocationContentCommand")
    .sc(GetTrustStoreRevocationContent$)
    .build() {
}

class ModifyCapacityReservationCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyCapacityReservation", {})
    .n("ElasticLoadBalancingV2Client", "ModifyCapacityReservationCommand")
    .sc(ModifyCapacityReservation$)
    .build() {
}

class ModifyIpPoolsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyIpPools", {})
    .n("ElasticLoadBalancingV2Client", "ModifyIpPoolsCommand")
    .sc(ModifyIpPools$)
    .build() {
}

class ModifyListenerAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyListenerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyListenerAttributesCommand")
    .sc(ModifyListenerAttributes$)
    .build() {
}

class ModifyListenerCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyListener", {})
    .n("ElasticLoadBalancingV2Client", "ModifyListenerCommand")
    .sc(ModifyListener$)
    .build() {
}

class ModifyLoadBalancerAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyLoadBalancerAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyLoadBalancerAttributesCommand")
    .sc(ModifyLoadBalancerAttributes$)
    .build() {
}

class ModifyRuleCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyRule", {})
    .n("ElasticLoadBalancingV2Client", "ModifyRuleCommand")
    .sc(ModifyRule$)
    .build() {
}

class ModifyTargetGroupAttributesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTargetGroupAttributes", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTargetGroupAttributesCommand")
    .sc(ModifyTargetGroupAttributes$)
    .build() {
}

class ModifyTargetGroupCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTargetGroup", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTargetGroupCommand")
    .sc(ModifyTargetGroup$)
    .build() {
}

class ModifyTrustStoreCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "ModifyTrustStore", {})
    .n("ElasticLoadBalancingV2Client", "ModifyTrustStoreCommand")
    .sc(ModifyTrustStore$)
    .build() {
}

class RegisterTargetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RegisterTargets", {})
    .n("ElasticLoadBalancingV2Client", "RegisterTargetsCommand")
    .sc(RegisterTargets$)
    .build() {
}

class RemoveListenerCertificatesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveListenerCertificates", {})
    .n("ElasticLoadBalancingV2Client", "RemoveListenerCertificatesCommand")
    .sc(RemoveListenerCertificates$)
    .build() {
}

class RemoveTagsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveTags", {})
    .n("ElasticLoadBalancingV2Client", "RemoveTagsCommand")
    .sc(RemoveTags$)
    .build() {
}

class RemoveTrustStoreRevocationsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "RemoveTrustStoreRevocations", {})
    .n("ElasticLoadBalancingV2Client", "RemoveTrustStoreRevocationsCommand")
    .sc(RemoveTrustStoreRevocations$)
    .build() {
}

class SetIpAddressTypeCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetIpAddressType", {})
    .n("ElasticLoadBalancingV2Client", "SetIpAddressTypeCommand")
    .sc(SetIpAddressType$)
    .build() {
}

class SetRulePrioritiesCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetRulePriorities", {})
    .n("ElasticLoadBalancingV2Client", "SetRulePrioritiesCommand")
    .sc(SetRulePriorities$)
    .build() {
}

class SetSecurityGroupsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetSecurityGroups", {})
    .n("ElasticLoadBalancingV2Client", "SetSecurityGroupsCommand")
    .sc(SetSecurityGroups$)
    .build() {
}

class SetSubnetsCommand extends Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ElasticLoadBalancing_v10", "SetSubnets", {})
    .n("ElasticLoadBalancingV2Client", "SetSubnetsCommand")
    .sc(SetSubnets$)
    .build() {
}

const paginateDescribeAccountLimits = createPaginator(ElasticLoadBalancingV2Client, DescribeAccountLimitsCommand, "Marker", "NextMarker", "");

const paginateDescribeListenerCertificates = createPaginator(ElasticLoadBalancingV2Client, DescribeListenerCertificatesCommand, "Marker", "NextMarker", "");

const paginateDescribeListeners = createPaginator(ElasticLoadBalancingV2Client, DescribeListenersCommand, "Marker", "NextMarker", "");

const paginateDescribeLoadBalancers = createPaginator(ElasticLoadBalancingV2Client, DescribeLoadBalancersCommand, "Marker", "NextMarker", "");

const paginateDescribeRules = createPaginator(ElasticLoadBalancingV2Client, DescribeRulesCommand, "Marker", "NextMarker", "");

const paginateDescribeTargetGroups = createPaginator(ElasticLoadBalancingV2Client, DescribeTargetGroupsCommand, "Marker", "NextMarker", "");

const paginateDescribeTrustStoreAssociations = createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoreAssociationsCommand, "Marker", "NextMarker", "PageSize");

const paginateDescribeTrustStoreRevocations = createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoreRevocationsCommand, "Marker", "NextMarker", "PageSize");

const paginateDescribeTrustStores = createPaginator(ElasticLoadBalancingV2Client, DescribeTrustStoresCommand, "Marker", "NextMarker", "PageSize");

const checkState$4 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.LoadBalancers);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "active");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.LoadBalancers);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            for (let anyStringEq_4 of returnComparator()) {
                if (anyStringEq_4 == "provisioning") {
                    return { state: WaiterState.RETRY, reason };
                }
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "LoadBalancerNotFoundException") {
            return { state: WaiterState.RETRY, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForLoadBalancerAvailable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
};
const waitUntilLoadBalancerAvailable = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$4);
    return checkExceptions(result);
};

const checkState$3 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        return { state: WaiterState.SUCCESS, reason };
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "LoadBalancerNotFoundException") {
            return { state: WaiterState.RETRY, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForLoadBalancerExists = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
};
const waitUntilLoadBalancerExists = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$3);
    return checkExceptions(result);
};

const checkState$2 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeLoadBalancersCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.LoadBalancers);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.State.Code;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "active");
            }
            if (allStringEq_5) {
                return { state: WaiterState.RETRY, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "LoadBalancerNotFoundException") {
            return { state: WaiterState.SUCCESS, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForLoadBalancersDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
};
const waitUntilLoadBalancersDeleted = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$2);
    return checkExceptions(result);
};

const checkState$1 = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeTargetHealthCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.TargetHealthDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.TargetHealth.State;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "unused");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "InvalidTargetException") {
            return { state: WaiterState.SUCCESS, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForTargetDeregistered = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
};
const waitUntilTargetDeregistered = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState$1);
    return checkExceptions(result);
};

const checkState = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new DescribeTargetHealthCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                let flat_1 = [].concat(...result.TargetHealthDescriptions);
                let projection_3 = flat_1.map((element_2) => {
                    return element_2.TargetHealth.State;
                });
                return projection_3;
            };
            let allStringEq_5 = (returnComparator().length > 0);
            for (let element_4 of returnComparator()) {
                allStringEq_5 = allStringEq_5 && (element_4 == "healthy");
            }
            if (allStringEq_5) {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
        if (exception.name === "InvalidInstance") {
            return { state: WaiterState.RETRY, reason };
        }
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForTargetInService = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilTargetInService = async (params, input) => {
    const serviceDefaults = { minDelay: 15, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return checkExceptions(result);
};

const commands = {
    AddListenerCertificatesCommand,
    AddTagsCommand,
    AddTrustStoreRevocationsCommand,
    CreateListenerCommand,
    CreateLoadBalancerCommand,
    CreateRuleCommand,
    CreateTargetGroupCommand,
    CreateTrustStoreCommand,
    DeleteListenerCommand,
    DeleteLoadBalancerCommand,
    DeleteRuleCommand,
    DeleteSharedTrustStoreAssociationCommand,
    DeleteTargetGroupCommand,
    DeleteTrustStoreCommand,
    DeregisterTargetsCommand,
    DescribeAccountLimitsCommand,
    DescribeCapacityReservationCommand,
    DescribeListenerAttributesCommand,
    DescribeListenerCertificatesCommand,
    DescribeListenersCommand,
    DescribeLoadBalancerAttributesCommand,
    DescribeLoadBalancersCommand,
    DescribeRulesCommand,
    DescribeSSLPoliciesCommand,
    DescribeTagsCommand,
    DescribeTargetGroupAttributesCommand,
    DescribeTargetGroupsCommand,
    DescribeTargetHealthCommand,
    DescribeTrustStoreAssociationsCommand,
    DescribeTrustStoreRevocationsCommand,
    DescribeTrustStoresCommand,
    GetResourcePolicyCommand,
    GetTrustStoreCaCertificatesBundleCommand,
    GetTrustStoreRevocationContentCommand,
    ModifyCapacityReservationCommand,
    ModifyIpPoolsCommand,
    ModifyListenerCommand,
    ModifyListenerAttributesCommand,
    ModifyLoadBalancerAttributesCommand,
    ModifyRuleCommand,
    ModifyTargetGroupCommand,
    ModifyTargetGroupAttributesCommand,
    ModifyTrustStoreCommand,
    RegisterTargetsCommand,
    RemoveListenerCertificatesCommand,
    RemoveTagsCommand,
    RemoveTrustStoreRevocationsCommand,
    SetIpAddressTypeCommand,
    SetRulePrioritiesCommand,
    SetSecurityGroupsCommand,
    SetSubnetsCommand,
};
const paginators = {
    paginateDescribeAccountLimits,
    paginateDescribeListenerCertificates,
    paginateDescribeListeners,
    paginateDescribeLoadBalancers,
    paginateDescribeRules,
    paginateDescribeTargetGroups,
    paginateDescribeTrustStoreAssociations,
    paginateDescribeTrustStoreRevocations,
    paginateDescribeTrustStores,
};
const waiters = {
    waitUntilLoadBalancerAvailable,
    waitUntilLoadBalancerExists,
    waitUntilLoadBalancersDeleted,
    waitUntilTargetDeregistered,
    waitUntilTargetInService,
};
class ElasticLoadBalancingV2 extends ElasticLoadBalancingV2Client {
}
createAggregatedClient(commands, ElasticLoadBalancingV2, { paginators, waiters });

const AuthenticateCognitoActionConditionalBehaviorEnum = {
    ALLOW: "allow",
    AUTHENTICATE: "authenticate",
    DENY: "deny",
};
const AuthenticateOidcActionConditionalBehaviorEnum = {
    ALLOW: "allow",
    AUTHENTICATE: "authenticate",
    DENY: "deny",
};
const JwtValidationActionAdditionalClaimFormatEnum = {
    SINGLE_STRING: "single-string",
    SPACE_SEPARATED_VALUES: "space-separated-values",
    STRING_ARRAY: "string-array",
};
const RedirectActionStatusCodeEnum = {
    HTTP_301: "HTTP_301",
    HTTP_302: "HTTP_302",
};
const ActionTypeEnum = {
    AUTHENTICATE_COGNITO: "authenticate-cognito",
    AUTHENTICATE_OIDC: "authenticate-oidc",
    FIXED_RESPONSE: "fixed-response",
    FORWARD: "forward",
    JWT_VALIDATION: "jwt-validation",
    REDIRECT: "redirect",
};
const RevocationType = {
    CRL: "CRL",
};
const TargetAdministrativeOverrideReasonEnum = {
    INTERNAL_ERROR: "AdministrativeOverride.Unknown",
    NO_OVERRIDE_ENGAGED: "AdministrativeOverride.NoOverride",
    ZONAL_SHIFT_DELEGATED_TO_DNS: "AdministrativeOverride.ZonalShiftDelegatedToDns",
    ZONAL_SHIFT_ENGAGED: "AdministrativeOverride.ZonalShiftActive",
};
const TargetAdministrativeOverrideStateEnum = {
    NO_OVERRIDE: "no_override",
    UNKNOWN: "unknown",
    ZONAL_SHIFT_ACTIVE: "zonal_shift_active",
    ZONAL_SHIFT_DELEGATED_TO_DNS: "zonal_shift_delegated_to_dns",
};
const AdvertiseTrustStoreCaNamesEnum = {
    off: "off",
    on: "on",
};
const MitigationInEffectEnum = {
    NO: "no",
    YES: "yes",
};
const AnomalyResultEnum = {
    ANOMALOUS: "anomalous",
    NORMAL: "normal",
};
const CapacityReservationStateEnum = {
    FAILED: "failed",
    PENDING: "pending",
    PROVISIONED: "provisioned",
    REBALANCING: "rebalancing",
};
const TrustStoreAssociationStatusEnum = {
    ACTIVE: "active",
    REMOVED: "removed",
};
const ProtocolEnum = {
    GENEVE: "GENEVE",
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    QUIC: "QUIC",
    TCP: "TCP",
    TCP_QUIC: "TCP_QUIC",
    TCP_UDP: "TCP_UDP",
    TLS: "TLS",
    UDP: "UDP",
};
const EnablePrefixForIpv6SourceNatEnum = {
    OFF: "off",
    ON: "on",
};
const IpAddressType = {
    DUALSTACK: "dualstack",
    DUALSTACK_WITHOUT_PUBLIC_IPV4: "dualstack-without-public-ipv4",
    IPV4: "ipv4",
};
const LoadBalancerSchemeEnum = {
    INTERNAL: "internal",
    INTERNET_FACING: "internet-facing",
};
const LoadBalancerTypeEnum = {
    APPLICATION: "application",
    GATEWAY: "gateway",
    NETWORK: "network",
};
const LoadBalancerStateEnum = {
    ACTIVE: "active",
    ACTIVE_IMPAIRED: "active_impaired",
    FAILED: "failed",
    PROVISIONING: "provisioning",
};
const TransformTypeEnum = {
    HOST_HEADER_REWRITE: "host-header-rewrite",
    URL_REWRITE: "url-rewrite",
};
const TargetGroupIpAddressTypeEnum = {
    IPV4: "ipv4",
    IPV6: "ipv6",
};
const TargetTypeEnum = {
    ALB: "alb",
    INSTANCE: "instance",
    IP: "ip",
    LAMBDA: "lambda",
};
const TrustStoreStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
};
const DescribeTargetHealthInputIncludeEnum = {
    ALL: "All",
    ANOMALY: "AnomalyDetection",
};
const TargetHealthReasonEnum = {
    DEREGISTRATION_IN_PROGRESS: "Target.DeregistrationInProgress",
    FAILED_HEALTH_CHECKS: "Target.FailedHealthChecks",
    HEALTH_CHECK_DISABLED: "Target.HealthCheckDisabled",
    INITIAL_HEALTH_CHECKING: "Elb.InitialHealthChecking",
    INTERNAL_ERROR: "Elb.InternalError",
    INVALID_STATE: "Target.InvalidState",
    IP_UNUSABLE: "Target.IpUnusable",
    NOT_IN_USE: "Target.NotInUse",
    NOT_REGISTERED: "Target.NotRegistered",
    REGISTRATION_IN_PROGRESS: "Elb.RegistrationInProgress",
    RESPONSE_CODE_MISMATCH: "Target.ResponseCodeMismatch",
    TIMEOUT: "Target.Timeout",
};
const TargetHealthStateEnum = {
    DRAINING: "draining",
    HEALTHY: "healthy",
    INITIAL: "initial",
    UNAVAILABLE: "unavailable",
    UNHEALTHY: "unhealthy",
    UNHEALTHY_DRAINING: "unhealthy.draining",
    UNUSED: "unused",
};
const RemoveIpamPoolEnum = {
    ipv4: "ipv4",
};
const EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum = {
    off: "off",
    on: "on",
};

exports.ActionTypeEnum = ActionTypeEnum;
exports.AddListenerCertificatesCommand = AddListenerCertificatesCommand;
exports.AddTagsCommand = AddTagsCommand;
exports.AddTrustStoreRevocationsCommand = AddTrustStoreRevocationsCommand;
exports.AdvertiseTrustStoreCaNamesEnum = AdvertiseTrustStoreCaNamesEnum;
exports.AnomalyResultEnum = AnomalyResultEnum;
exports.AuthenticateCognitoActionConditionalBehaviorEnum = AuthenticateCognitoActionConditionalBehaviorEnum;
exports.AuthenticateOidcActionConditionalBehaviorEnum = AuthenticateOidcActionConditionalBehaviorEnum;
exports.CapacityReservationStateEnum = CapacityReservationStateEnum;
exports.CreateListenerCommand = CreateListenerCommand;
exports.CreateLoadBalancerCommand = CreateLoadBalancerCommand;
exports.CreateRuleCommand = CreateRuleCommand;
exports.CreateTargetGroupCommand = CreateTargetGroupCommand;
exports.CreateTrustStoreCommand = CreateTrustStoreCommand;
exports.DeleteListenerCommand = DeleteListenerCommand;
exports.DeleteLoadBalancerCommand = DeleteLoadBalancerCommand;
exports.DeleteRuleCommand = DeleteRuleCommand;
exports.DeleteSharedTrustStoreAssociationCommand = DeleteSharedTrustStoreAssociationCommand;
exports.DeleteTargetGroupCommand = DeleteTargetGroupCommand;
exports.DeleteTrustStoreCommand = DeleteTrustStoreCommand;
exports.DeregisterTargetsCommand = DeregisterTargetsCommand;
exports.DescribeAccountLimitsCommand = DescribeAccountLimitsCommand;
exports.DescribeCapacityReservationCommand = DescribeCapacityReservationCommand;
exports.DescribeListenerAttributesCommand = DescribeListenerAttributesCommand;
exports.DescribeListenerCertificatesCommand = DescribeListenerCertificatesCommand;
exports.DescribeListenersCommand = DescribeListenersCommand;
exports.DescribeLoadBalancerAttributesCommand = DescribeLoadBalancerAttributesCommand;
exports.DescribeLoadBalancersCommand = DescribeLoadBalancersCommand;
exports.DescribeRulesCommand = DescribeRulesCommand;
exports.DescribeSSLPoliciesCommand = DescribeSSLPoliciesCommand;
exports.DescribeTagsCommand = DescribeTagsCommand;
exports.DescribeTargetGroupAttributesCommand = DescribeTargetGroupAttributesCommand;
exports.DescribeTargetGroupsCommand = DescribeTargetGroupsCommand;
exports.DescribeTargetHealthCommand = DescribeTargetHealthCommand;
exports.DescribeTargetHealthInputIncludeEnum = DescribeTargetHealthInputIncludeEnum;
exports.DescribeTrustStoreAssociationsCommand = DescribeTrustStoreAssociationsCommand;
exports.DescribeTrustStoreRevocationsCommand = DescribeTrustStoreRevocationsCommand;
exports.DescribeTrustStoresCommand = DescribeTrustStoresCommand;
exports.ElasticLoadBalancingV2 = ElasticLoadBalancingV2;
exports.ElasticLoadBalancingV2Client = ElasticLoadBalancingV2Client;
exports.EnablePrefixForIpv6SourceNatEnum = EnablePrefixForIpv6SourceNatEnum;
exports.EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum = EnforceSecurityGroupInboundRulesOnPrivateLinkTrafficEnum;
exports.GetResourcePolicyCommand = GetResourcePolicyCommand;
exports.GetTrustStoreCaCertificatesBundleCommand = GetTrustStoreCaCertificatesBundleCommand;
exports.GetTrustStoreRevocationContentCommand = GetTrustStoreRevocationContentCommand;
exports.IpAddressType = IpAddressType;
exports.JwtValidationActionAdditionalClaimFormatEnum = JwtValidationActionAdditionalClaimFormatEnum;
exports.LoadBalancerSchemeEnum = LoadBalancerSchemeEnum;
exports.LoadBalancerStateEnum = LoadBalancerStateEnum;
exports.LoadBalancerTypeEnum = LoadBalancerTypeEnum;
exports.MitigationInEffectEnum = MitigationInEffectEnum;
exports.ModifyCapacityReservationCommand = ModifyCapacityReservationCommand;
exports.ModifyIpPoolsCommand = ModifyIpPoolsCommand;
exports.ModifyListenerAttributesCommand = ModifyListenerAttributesCommand;
exports.ModifyListenerCommand = ModifyListenerCommand;
exports.ModifyLoadBalancerAttributesCommand = ModifyLoadBalancerAttributesCommand;
exports.ModifyRuleCommand = ModifyRuleCommand;
exports.ModifyTargetGroupAttributesCommand = ModifyTargetGroupAttributesCommand;
exports.ModifyTargetGroupCommand = ModifyTargetGroupCommand;
exports.ModifyTrustStoreCommand = ModifyTrustStoreCommand;
exports.ProtocolEnum = ProtocolEnum;
exports.RedirectActionStatusCodeEnum = RedirectActionStatusCodeEnum;
exports.RegisterTargetsCommand = RegisterTargetsCommand;
exports.RemoveIpamPoolEnum = RemoveIpamPoolEnum;
exports.RemoveListenerCertificatesCommand = RemoveListenerCertificatesCommand;
exports.RemoveTagsCommand = RemoveTagsCommand;
exports.RemoveTrustStoreRevocationsCommand = RemoveTrustStoreRevocationsCommand;
exports.RevocationType = RevocationType;
exports.SetIpAddressTypeCommand = SetIpAddressTypeCommand;
exports.SetRulePrioritiesCommand = SetRulePrioritiesCommand;
exports.SetSecurityGroupsCommand = SetSecurityGroupsCommand;
exports.SetSubnetsCommand = SetSubnetsCommand;
exports.TargetAdministrativeOverrideReasonEnum = TargetAdministrativeOverrideReasonEnum;
exports.TargetAdministrativeOverrideStateEnum = TargetAdministrativeOverrideStateEnum;
exports.TargetGroupIpAddressTypeEnum = TargetGroupIpAddressTypeEnum;
exports.TargetHealthReasonEnum = TargetHealthReasonEnum;
exports.TargetHealthStateEnum = TargetHealthStateEnum;
exports.TargetTypeEnum = TargetTypeEnum;
exports.TransformTypeEnum = TransformTypeEnum;
exports.TrustStoreAssociationStatusEnum = TrustStoreAssociationStatusEnum;
exports.TrustStoreStatus = TrustStoreStatus;
exports.paginateDescribeAccountLimits = paginateDescribeAccountLimits;
exports.paginateDescribeListenerCertificates = paginateDescribeListenerCertificates;
exports.paginateDescribeListeners = paginateDescribeListeners;
exports.paginateDescribeLoadBalancers = paginateDescribeLoadBalancers;
exports.paginateDescribeRules = paginateDescribeRules;
exports.paginateDescribeTargetGroups = paginateDescribeTargetGroups;
exports.paginateDescribeTrustStoreAssociations = paginateDescribeTrustStoreAssociations;
exports.paginateDescribeTrustStoreRevocations = paginateDescribeTrustStoreRevocations;
exports.paginateDescribeTrustStores = paginateDescribeTrustStores;
exports.waitForLoadBalancerAvailable = waitForLoadBalancerAvailable;
exports.waitForLoadBalancerExists = waitForLoadBalancerExists;
exports.waitForLoadBalancersDeleted = waitForLoadBalancersDeleted;
exports.waitForTargetDeregistered = waitForTargetDeregistered;
exports.waitForTargetInService = waitForTargetInService;
exports.waitUntilLoadBalancerAvailable = waitUntilLoadBalancerAvailable;
exports.waitUntilLoadBalancerExists = waitUntilLoadBalancerExists;
exports.waitUntilLoadBalancersDeleted = waitUntilLoadBalancersDeleted;
exports.waitUntilTargetDeregistered = waitUntilTargetDeregistered;
exports.waitUntilTargetInService = waitUntilTargetInService;
